daemon/container_unix.go

```diff
@@ -810,6 +812,17 @@ func (container *Container) buildCreateEndpointOptions(n libnetwork.Network) ([]
 		createOptions []libnetwork.EndpointOption
 	)
 
+	if n.Name() == "bridge" || container.NetworkSettings.IsAnonymousEndpoint {
+		createOptions = append(createOptions, libnetwork.CreateOptionAnonymous())
+	}
+
+	// Other configs are applicable only for the endpoint in the network
+	// to which container was connected to on docker run.
+	if n.Name() != container.hostConfig.NetworkMode.NetworkName() &&
+		n.Name() != "bridge" {
```

Comment:
@mavenugo  `container.hostConfig.NetworkMode.IsDefault() && n.Name() == "bridge"` should be ok, avoid passing the controller, see https://github.com/docker/docker/pull/17860/files#r44446197 from @aboch , `bridge` is hard code